pallets/dday-protocol/src/lib.rs

```diff
@@ -0,0 +1,169 @@
+#![cfg_attr(not(feature = "std"), no_std)]
+
+#[cfg(test)]
+mod tests;
+
+mod default_weights;
+mod types;
+
+use default_weights::WeightInfo;
+use frame_support::pallet_prelude::DispatchResult;
+use frame_support::transactional;
+use frame_system::ensure_signed;
+use sp_runtime::DispatchError;
+use ternoa_common::traits::{CapsulesTrait, NFTTrait};
+use types::TransmissionData;
+
+pub use pallet::*;
+
+#[frame_support::pallet]
+pub mod pallet {
+    use super::*;
+
+    use frame_support::pallet_prelude::*;
+    use frame_system::pallet_prelude::{BlockNumberFor, OriginFor};
+    use ternoa_common::traits::{NFTTrait, CapsulesTrait};
+    use ternoa_primitives::{nfts::NFTId, BlockNumber};
+
+    #[pallet::pallet]
+    #[pallet::generate_store(pub(super) trait Store)]
+    pub struct Pallet<T>(_);
+
+    #[pallet::config]
+    pub trait Config: frame_system::Config {
+        // Because this pallet emits events, it depends on the runtime's definition of an event.
+        type Event: From<Event<Self>> + IsType<<Self as frame_system::Config>::Event>;
+        // Pallet managing NFTs.
+        type NFTs: NFTTrait<AccountId = Self::AccountId>;
+        // Capsules Pallet
+        type Capsules: CapsulesTrait<AccountId = Self::AccountId>;
+        // Weight values for this pallet
+        type WeightInfo: WeightInfo;
+    }
+
+    #[pallet::storage]
+    #[pallet::getter(fn transmissions)]
+    pub type Transmissions<T: Config> =
+        StorageMap<_, Blake2_128Concat, NFTId, TransmissionData<T::AccountId>, OptionQuery>;
+
+    #[pallet::genesis_config]
+    pub struct GenesisConfig {}
+
+    #[cfg(feature = "std")]
+    impl Default for GenesisConfig {
+        fn default() -> Self {
+            Self {}
+        }
+    }
+
+    #[pallet::genesis_build]
+    impl<T: Config> GenesisBuild<T> for GenesisConfig {
+        fn build(&self) {}
+    }
+
+    #[pallet::event]
+    #[pallet::generate_deposit(pub(super) fn deposit_event)]
+    pub enum Event<T: Config> {
+        // Transmission has been store
+        TransmissionStored,
+        // Transmission has been completed
+        TransmissionCompleted,
+        // Transmission has been cancelled
+        TransmissionCancelled,
```

Comment:
Add data to those events: Nft/CapsuleId, + ddate and recipient for TransmissionStored